image-annotator/src/redux/reducers/__tests__/faces.test.js

```diff
@@ -0,0 +1,235 @@
+import facesReducer from '../faces';
+
+// Import action labels & action creators for use in specs
+import * as actions from '../../actions';
+
+describe('faces reducer', () => {
+
+  it('is a function', () => {
+    expect(facesReducer).toBeDefined();
+    expect(facesReducer).toBeInstanceOf(Function);
+  });
+
+  it('initializes a default state object', () => {
+    const initialState = facesReducer(undefined, {});
+    expect(initialState).toEqual({
+      list: [],
+      selected: null,
+    });
+  });
+
+  it('does not mutate state when receiving an irrelevant action', () => {
+    const initialState = facesReducer(undefined, {});
+    const nextState = facesReducer(initialState, {});
+    expect(nextState).toBe(initialState);
+  });
+
+  const { RECEIVE_FACES } = actions;
+  describe(`on ${RECEIVE_FACES}`, () => {
+
+    it('sets the action\'s array payload as the faces property', () => {
+      const initialState = facesReducer(undefined, {});
+      const nextState = facesReducer(initialState, {
+        type: RECEIVE_FACES,
+        payload: [{ id: 'a' }, { id: 'b' }, { id: 'c' }],
+      });
+      expect(nextState).not.toBe(initialState);
+      expect(nextState.list).toBeDefined();
+      expect(nextState.list).not.toBe(initialState.list);
+      expect(nextState.list).toEqual([{ id: 'a' }, { id: 'b' }, { id: 'c' }]);
+    });
+
+    it('auto-selects the first face in the set', () => {
+      const initialState = facesReducer(undefined, {});
+      expect(initialState.selected).toBeNull();
+      const nextState = facesReducer(initialState, {
+        type: RECEIVE_FACES,
+        payload: [{ id: 'a' }, { id: 'b' }, { id: 'c' }],
+      });
+      expect(nextState.selected).toBe(0);
+    });
```

Comment:
@gnarf Do you have any patterns you particularly like for testing reducers? Referring specifically to how to set up the state and register the change (using existing action creators vs just providing the actions as they'd be received, etc)